astropy/modeling/fitting.py

```diff
@@ -1181,81 +1280,81 @@ def __call__(self, model, x, y, z=None, weights=None,
+    The ``method`` determines which fitter algorithm to use:
+        ‘trf’ : Trust Region Reflective algorithm, particularly suitable
+            for large sparse problems with bounds. Generally robust method.
+        ‘dogbox’ : dogleg algorithm with rectangular trust regions, typical
+            use case is small problems with bounds. Not recommended for
+            problems with rank-deficient Jacobian.
+        ‘lm’ : Levenberg-Marquardt algorithm as implemented in MINPACK.
+            Doesn’t handle bounds and sparse Jacobians. Usually the most
+            efficient method for small unconstrained problems.
+    """
 
-            pars = [getattr(model, name) for name in model.param_names]
-            fixed = [par.fixed for par in pars]
-            tied = [par.tied for par in pars]
-            tied = list(np.where([par.tied is not False for par in pars],
-                                 True, tied))
-            fix_and_tie = np.logical_or(fixed, tied)
-            ind = np.logical_not(fix_and_tie)
+    def __init__(self, calc_uncertainties=False, method='trf', check_bounds=False):
+        super().__init__(calc_uncertainties, check_bounds)
+        self._method = method
 
-            if not model.col_fit_deriv:
-                residues = np.asarray(full_deriv[np.nonzero(ind)]).T
-            else:
-                residues = full_deriv[np.nonzero(ind)]
+    def _run_fitter(self, model, farg, maxiter, acc, epsilon, estimate_jacobian):
+        from scipy import optimize
+        from scipy.linalg import svd
 
-            return [np.ravel(_) for _ in residues]
+        if model.fit_deriv is None or estimate_jacobian:
+            dfunc = '2-point'
         else:
-            if z is None:
-                try:
-                    return np.array([np.ravel(_) for _ in np.array(weights) *
-                                     np.array(model.fit_deriv(x, *params))])
-                except ValueError:
-                    return np.array([np.ravel(_) for _ in np.array(weights) *
-                                     np.moveaxis(
-                                         np.array(model.fit_deriv(x, *params)),
-                                         -1, 0)]).transpose()
-            else:
-                if not model.col_fit_deriv:
-                    return [np.ravel(_) for _ in
-                            (np.ravel(weights) * np.array(model.fit_deriv(x, y, *params)).T).T]
-                return [np.ravel(_) for _ in weights * np.array(model.fit_deriv(x, y, *params))]
+            def _dfunc(params, model, weights, x, y, z=None):
+                if model.col_fit_deriv:
+                    return np.transpose(self._wrap_deriv(params, model, weights, x, y, z))
+                else:
+                    return self._wrap_deriv(params, model, weights, x, y, z)
+
+            dfunc = _dfunc
+
+        init_values, _, bounds = model_to_fit_params(model)
+
+        if self._check_bounds:
+            bounds = (-np.inf, np.inf)
```

Comment:
I am a bit confused about this statement. Why are bounds set to values which in other fitters would indicate not to check them?

Reply:
I think I see now what this is doing but it may need a comment to help the next one reading the code.

Reply:
See cbe5e83